src/main/java/com/softlayer/api/Mask.java

```diff
@@ -44,7 +45,26 @@ protected String getMask() {
 
     @Override
     public String toString() {
-        return toString(new StringBuilder()).toString();
+        String objectMask = new String();
+        String builtedMask = toString(new StringBuilder()).toString();
+        if(builtedMask.contains("[")){
```

Comment:
I don't really understand what this code is trying to do, can you provide an example Mask that would trigger the `if(builtedMask.contains("[")){` branch and the `else` branch?

Reply:
I´m using this example code:

Account.Service accountService = Account.service(client);

accountService.withMask().blockDeviceTemplateGroups().id();
accountService.withMask().blockDeviceTemplateGroups().blockDevices().diskImage().type();
accountService.withMask().blockDeviceTemplateGroups().status().keyName();

List<Group> result = accountService.getBlockDeviceTemplateGroups();

For the conditional if(builtedMask.contains("["))is when I use more than one mask like above objectMask example.

The method toString() of the class Mask return the value like this:

blockDeviceTemplateGroups[id, blockDevices.diskImage.type, status.keyName]

with this data the objectMask will be building to send the request, when the request is sending like this without to put the getObject option between the accountService and  getBlockDeviceTemplateGroups() method the response will be that the blockDeviceTemplateGroups is not recognized as a value for the objectMask and the service. It is for that I implemented this code to remove the value blockDeviceTemplateGroups to send the correct objectMask without using getObject for the method getBlockDeviceTemplateGroups().

The result for the conditional if() will be:

id, blockDevices.diskImage.type, status.keyName

For the else branch is for one mask, like this example:

accountService.withMask().blockDeviceTemplateGroups().status().keyName();

The method toString() will return like this:

blockDeviceTemplateGroups.status.keyName

And with this branch else the result will be:

status.keyName

Reply:
Thanks, that helps me understand what is going on a lot better.

Reply:
I'm glad to hear that, now we can do the request using objectMask and resultLimit at the same time.